### ISSUE TITLE
s3: better error on missing credentials

### ISSUE THREAD

**Issue by colllin** [iterative/dvc#4478]
A colleague was recently onboarding to our dvc project and received this error:
```
$ dvc pull
ERROR: unexpected error - Unable to locate credentials
```

I believe that if this was a better error message, they may have figured it out on their own.  It wasn't clear that they were missing **AWS** credentials when running a **dvc** command.  Maybe you could say something more like "Unable to access s3://... (the remote url).  Do you have a network connection and have the proper credentials set up?"  which (I think) would address the problem more generally for various remote providers while giving a little more information to the user to figure it out.

**Comment by efiop** [comment]
@colllin Thanks for the feedback! Indeed, we could just wrap that exception and present something nicer. Let me see...

**Comment by efiop** [comment]
@colllin we've added some friendlier messages for s3, the change will be released in 1.6.2 today. Thanks for the feedback! :pray:

**Comment by colllin** [comment]
You’re awesome! Thank you!! 🎉
